services/src/tpu_utils/tpu_service.rs

```diff
@@ -163,48 +163,42 @@ impl TpuService {
             .await;
     }
 
-    fn check_exit_signal(exit_signal: &Arc<AtomicBool>) -> bool {
-        exit_signal.load(Ordering::Relaxed)
-    }
-
     async fn update_current_slot(
         &self,
         update_notifier: tokio::sync::mpsc::UnboundedSender<u64>,
-        exit_signal: Arc<AtomicBool>,
-    ) {
+    ) -> AnyhowJoinHandle {
         let current_slot = self.current_slot.clone();
-        let update_slot = |slot: u64| {
+        let rpc_client = self.rpc_client.clone();
+        let rpc_ws_address = self.rpc_ws_address.clone();
+
+        let update_slot = move |slot: u64| {
             if slot > current_slot.load(Ordering::Relaxed) {
                 current_slot.store(slot, Ordering::Relaxed);
                 CURRENT_SLOT.set(slot as i64);
                 let _ = update_notifier.send(slot);
             }
         };
-        let mut nb_errror = 0;
-        loop {
-            if Self::check_exit_signal(&exit_signal) {
-                break;
-            }
-            // always loop update the current slots as it is central to working of TPU
-            if let Err(e) =
-                SolanaUtils::poll_slots(self.rpc_client.clone(), &self.rpc_ws_address, update_slot)
-                    .await
-            {
+
+        tokio::spawn(async move {
+            let mut nb_errror = 0;
+
+            while nb_errror < MAX_NB_ERRORS {
+                // always loop update the current slots as it is central to working of TPU
+                let Err(err) = SolanaUtils::poll_slots(&rpc_client, &rpc_ws_address, &update_slot).await else {
+                    nb_errror = 0;
+                    continue;
+                };
+
                 nb_errror += 1;
-                log::info!("Got error while polling slot {}", e);
-                if nb_errror > MAX_NB_ERRORS {
-                    error!(
-                        "Reached max amount of errors to fetch latest slot, exiting poll slot loop"
-                    );
-                    break;
-                }
-            } else {
-                nb_errror = 0;
+                log::info!("Got error while polling slot {}", err);
             }
-        }
+
+            bail!("Reached max amount of errors to fetch latest slot, exiting poll slot loop")
```

Comment:
Should also log an error! here

Reply:
You are sure right?

Reply:
I prefer it having it twice rather than none because something went wrong with tokio stuff. Lite-rpc had problems in past where it went down without showing any logs which service went down.